apps/system/js/updatable.js

```diff
@@ -19,6 +21,49 @@
  * - cancelDownload() to cancel it
  */
 
+// Wrapper to read a settings value
+var setting = (function() {
+  var values = {};
+  return key => {
+
+    // Cache the promise straight away so multiple callers dont
+    // create multiple listeners
+    if (key in values) {
+      return values[key];
+    }
+
+    values[key] = new Promise(resolve => {
+      // We store the value of the setting and the observer is
+      // in charge of keeping it up to date
+      function cacheAndResolve() {
+        values[key] = (key in req.result) ? req.result[key] : undefined;
+        resolve(values[key]);
+      }
+
+      var lock = navigator.mozSettings.createLock();
+      var req = lock.get(key);
+
+      req.onerror = cacheAndResolve.bind(this);
```

Comment:
Are you sure you want to hide errors here? Seems better to reject the promise, and handle the rejection with a catch below. Right now we are silently disabling auto updates and add-on notifications. Not sure this is what we want for things like Settings db errros.